### PR TITLE
more graceful docgen

### PR DESCRIPTION
**Pull request by ajm188**
skip programs who either don't have a target directory (in content/en/.../) for that version, or don't have a docgen directory (go/cmd/<binary>/docgen) in vitessio/vitess for that gitref

this allows us to freely bulk-generate docs for all programs/versions without having to get fussy about it

## Example Output

(note i have `upstream/main` checked out as `vmain` so i tweaked the makefile locally)

<details>
```
➜  vtwebsite git:(andrew/docgen-graceful) ✗ VITESS_DIR=~/dev/vitess make generated-docs         
go run ./tools/cobradocs/ --vitess-dir "/Users/andrew/dev/vitess" --version-pairs ""vmain:18.0,v17.0.2:17.0,v16.0.4:16.0,v15.0.4:15.0"" vtctld
2023/09/18 06:27:41 [warning] cannot find directory for doc version v17.0.2:17.0 (index=1): stat /Users/andrew/dev/vtwebsite/content/en/docs/17.0/reference/programs/vtctld: no such file or directory
2023/09/18 06:27:41 [warning] cannot find directory for doc version v16.0.4:16.0 (index=2): stat /Users/andrew/dev/vtwebsite/content/en/docs/16.0/reference/programs/vtctld: no such file or directory
2023/09/18 06:27:41 [warning] cannot find directory for doc version v15.0.4:15.0 (index=3): stat /Users/andrew/dev/vtwebsite/content/en/docs/15.0/reference/programs/vtctld: no such file or directory
go run ./tools/cobradocs/ --vitess-dir "/Users/andrew/dev/vitess" --version-pairs ""vmain:18.0,v17.0.2:17.0,v16.0.4:16.0,v15.0.4:15.0"" vtctldclient
go run ./tools/cobradocs/ --vitess-dir "/Users/andrew/dev/vitess" --version-pairs ""vmain:18.0,v17.0.2:17.0,v16.0.4:16.0,v15.0.4:15.0"" vtgate
2023/09/18 06:28:01 [warning] cannot find directory for doc version v17.0.2:17.0 (index=1): stat /Users/andrew/dev/vtwebsite/content/en/docs/17.0/reference/programs/vtgate: no such file or directory
2023/09/18 06:28:01 [warning] cannot find directory for doc version v16.0.4:16.0 (index=2): stat /Users/andrew/dev/vtwebsite/content/en/docs/16.0/reference/programs/vtgate: no such file or directory
2023/09/18 06:28:01 [warning] cannot find directory for doc version v15.0.4:15.0 (index=3): stat /Users/andrew/dev/vtwebsite/content/en/docs/15.0/reference/programs/vtgate: no such file or directory
go run ./tools/cobradocs/ --vitess-dir "/Users/andrew/dev/vitess" --version-pairs ""vmain:18.0,v17.0.2:17.0,v16.0.4:16.0,v15.0.4:15.0"" vtorc
2023/09/18 06:28:04 [warning] cannot find directory for doc version v17.0.2:17.0 (index=1): stat /Users/andrew/dev/vtwebsite/content/en/docs/17.0/reference/programs/vtorc: no such file or directory
2023/09/18 06:28:04 [warning] cannot find directory for doc version v16.0.4:16.0 (index=2): stat /Users/andrew/dev/vtwebsite/content/en/docs/16.0/reference/programs/vtorc: no such file or directory
2023/09/18 06:28:04 [warning] cannot find directory for doc version v15.0.4:15.0 (index=3): stat /Users/andrew/dev/vtwebsite/content/en/docs/15.0/reference/programs/vtorc: no such file or directory
go run ./tools/cobradocs/ --vitess-dir "/Users/andrew/dev/vitess" --version-pairs ""vmain:18.0,v17.0.2:17.0,v16.0.4:16.0,v15.0.4:15.0"" vttablet
2023/09/18 06:28:05 [warning] cannot find directory for doc version v17.0.2:17.0 (index=1): stat /Users/andrew/dev/vtwebsite/content/en/docs/17.0/reference/programs/vttablet: no such file or directory
2023/09/18 06:28:05 [warning] cannot find directory for doc version v16.0.4:16.0 (index=2): stat /Users/andrew/dev/vtwebsite/content/en/docs/16.0/reference/programs/vttablet: no such file or directory
2023/09/18 06:28:05 [warning] cannot find directory for doc version v15.0.4:15.0 (index=3): stat /Users/andrew/dev/vtwebsite/content/en/docs/15.0/reference/programs/vttablet: no such file or directory
➜  vtwebsite git:(andrew/docgen-graceful) ✗ echo $?
0
```

### CODE REVIEW

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md [hunk 1]
```diff
@@ -8,16 +8,16 @@ Perform commands related to moving tables from a source keyspace to a target key
 
 ### Synopsis
 
-MoveTables commands: Create, Show, Status, SwitchTraffic, ReverseTraffic, Stop, Start, Cancel, and Delete.
+moveTables commands: Create, Show, Status, SwitchTraffic, ReverseTraffic, Stop, Start, Cancel, and Delete.
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md:11]: Why did the case change on this?

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md [hunk 2]
```diff
@@ -32,10 +32,10 @@ See the --help output for each command for more details.
 * [vtctldclient](../)	 - Executes a cluster management command on the remote vtctld server.
 * [vtctldclient MoveTables cancel](./vtctldclient_movetables_cancel/)	 - Cancel a MoveTables VReplication workflow.
 * [vtctldclient MoveTables complete](./vtctldclient_movetables_complete/)	 - Complete a MoveTables VReplication workflow.
-* [vtctldclient MoveTables create](./vtctldclient_movetables_create/)	 - Create and optionally run a MoveTables VReplication workflow.
+* [vtctldclient MoveTables create](./vtctldclient_movetables_create/)	 - Create and optionally run a moveTables VReplication workflow.
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md:35]: Also here.

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/vtctldclient_MoveTables_create.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ series: vtctldclient
 ---
 ## vtctldclient MoveTables create
 
-Create and optionally run a MoveTables VReplication workflow.
+Create and optionally run a moveTables VReplication workflow.
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/vtctldclient_MoveTables_create.md:7]: And here

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/vtctldclient_MoveTables_create.md [hunk 4]
```diff
@@ -19,29 +19,29 @@ vtctldclient --server localhost:15999 movetables --workflow commerce2customer --
...
-      --source-time-zone string            Specifying this causes any DATETIME fields to be converted from the given time zone into UTC
-      --stop-after-copy                    Stop the MoveTables workflow after it's finished copying the existing rows and before it starts replicating changes
-      --tables strings                     Source tables to copy
-      --tablet-types strings               Source tablet types to replicate table data from (e.g. PRIMARY,REPLICA,RDONLY)
-      --tablet-types-in-preference-order   When performing source tablet selection, look for candidates in the type order as they are listed in the tablet-types flag (default true)
+      --on-ddl string                      What to do when DDL is encountered in the VReplication stream. Possible values are IGNORE, STOP, EXEC, and EXEC_IGNORE. (default "IGNORE")
+      --source-keyspace string             Keyspace where the tables are being moved from (required).
+      --source-shards strings              Source shards to copy data from when performing a partial moveTables (experimental).
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/vtctldclient_MoveTables_create.md:32]: And here.

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/_index.md [hunk 5]
```diff
@@ -22,12 +22,12 @@ Operates on online DDL (schema migrations).
...
-* [vtctldclient OnlineDDL launch](./vtctldclient_onlineddl_launch/)	 - Launch one or all migrations executed with `--postpone-launch`.
+* [vtctldclient OnlineDDL complete](./vtctldclient_onlineddl_complete/)	 - complete one or all migrations executed with --postpone-completion
+* [vtctldclient OnlineDDL launch](./vtctldclient_onlineddl_launch/)	 - launch one or all migrations executed with --postpone-launch
 * [vtctldclient OnlineDDL retry](./vtctldclient_onlineddl_retry/)	 - Mark a given schema migration for retry.
 * [vtctldclient OnlineDDL show](./vtctldclient_onlineddl_show/)	 - Display information about online DDL operations.
-* [vtctldclient OnlineDDL throttle](./vtctldclient_onlineddl_throttle/)	 - Throttle one or all migrations.
-* [vtctldclient OnlineDDL unthrottle](./vtctldclient_onlineddl_unthrottle/)	 - Unthrottle one or all migrations.
+* [vtctldclient OnlineDDL throttle](./vtctldclient_onlineddl_throttle/)	 - Throttles one or all migrations
+* [vtctldclient OnlineDDL unthrottle](./vtctldclient_onlineddl_unthrottle/)	 - Unthrottles one or all migrations
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/_index.md:32]: Inconsistent casing. Some start with upper, some with lower.

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/vtctldclient_OnlineDDL_cancel.md [hunk 6]
```diff
@@ -4,7 +4,7 @@ series: vtctldclient
 ---
 ## vtctldclient OnlineDDL cancel
 
-Cancel one or all pending migrations, terminating any actively running ones.
+cancel one or all migrations, terminating any running ones as needed.
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/vtctldclient_OnlineDDL_cancel.md:7]: My preference is `Cancel` vs. `cancel`.

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/vtctldclient_OnlineDDL_complete.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ series: vtctldclient
 ---
 ## vtctldclient OnlineDDL complete
 
-Complete one or all migrations executed with `--postpone-completion`.
+complete one or all migrations executed with --postpone-completion
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/vtctldclient_OnlineDDL_complete.md:7]: `Complete`.

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/vtctldclient_OnlineDDL_launch.md [hunk 8]
```diff
@@ -4,7 +4,7 @@ series: vtctldclient
 ---
 ## vtctldclient OnlineDDL launch
 
-Launch one or all migrations executed with `--postpone-launch`.
+launch one or all migrations executed with --postpone-launch
```
Comment (deepthi) [inline content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_OnlineDDL/vtctldclient_OnlineDDL_launch.md:7]: `Launch`.

**Review by deepthi — COMMENTED** [review]
All my comments are to do with lowercase vs uppercase. I believe they will need to change in the main repo and the docs have to be regenerated.

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md (hunk 1 above)
Comment (ajm188) [reply content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md:11]: https://github.com/vitessio/vitess/blob/main/go/cmd/vtctldclient/command/vreplication/movetables/movetables.go#L30

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md (hunk 2 above)
Comment (ajm188) [reply content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/_index.md:35]: https://github.com/vitessio/vitess/blob/main/go/cmd/vtctldclient/command/vreplication/movetables/create.go#L47

File: content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/vtctldclient_MoveTables_create.md (hunk 3 above)
Comment (ajm188) [reply content/en/docs/18.0/reference/programs/vtctldclient/vtctldclient_MoveTables/vtctldclient_MoveTables_create.md:7]: https://github.com/vitessio/vitess/blob/main/go/cmd/vtctldclient/command/vreplication/movetables/create.go#L47

**Comment by ajm188** [comment]
tl;dr all the casing changes were made in the main codebase but never regenerated back to the website docs. this is just correcting that

i’ll defer to @rohit-nayak-ps for what consistency he wants there (i.e. i’d like to get the main change in and follow up for the consistentcy)

**Review by deepthi — APPROVED** [review]
